packages/amplify-category-api/src/commands/api/add-graphql-datasource.js

```diff
@@ -105,16 +106,25 @@ module.exports = {
          */
         const apiDirPath = `${projectBackendDirPath}/${category}/${resourceName}`;
         fs.ensureDirSync(apiDirPath);
-        const graphqlSchemaFilePath = `${apiDirPath}/schema.graphql`;
-        fs.ensureFileSync(graphqlSchemaFilePath);
-        const graphqlSchemaRaw = fs.readFileSync(graphqlSchemaFilePath, 'utf8');
-        const currGraphQLSchemaDoc = graphql.parse(graphqlSchemaRaw);
-
+        const graphqlSchemaFilePath = path.join(apiDirPath, 'schema.graphql');
+        const schemaFileExists = fs.existsSync(graphqlSchemaFilePath);
         const rdsGraphQLSchemaDoc = graphqlSchemaContext.schemaDoc;
+        const schemaDirectoryPath = path.join(apiDirPath, 'schema');
+        const schemaDirectoryExists = fs.existsSync(schemaDirectoryPath);
+
+        if (schemaFileExists) {
+          const graphqlSchemaRaw = fs.readFileSync(graphqlSchemaFilePath).toString();
+          const currGraphQLSchemaDoc = graphql.parse(graphqlSchemaRaw);
```

Comment:
We need to handle the parsing errors, there could be 2 casese
1. empty file
2. Invalid schema

In case of empty file, we can swallow the error and just merge the content. There is also another possiblity where the type is already declared previously and we may want to handle by erroring out and saying to customer that the that type is already present. Doing a validation after merge before writing to disk so we dont write invalid schema

Reply:
Can we return the error that graphql.parse will return?

Reply:
We could, I am wondering if we should just return or wait for the customers to fix issues with their schema, like how when you add a graphql schema, if the schema has error we wait till the error is fixed. The reasoning behind this is customer might have answered a few question to get to this step and we probably dont want them to reapeat the same steps again

Reply:
Okay I understand. In above case customer edit the schema and wait till the error is fixed. But here customer already has a graphql schema either in api folder or schema folder. If its in API folder we are reading and merging with rds schema.
Either two cases are possible 
1) if the schema is valid then happy path. 
2) If not valid , then we can throw error at parse before merging with rds schema and writing invalid file to disk